driver-core/src/main/com/mongodb/internal/connection/ScramShaAuthenticator.java

```diff
@@ -132,7 +140,7 @@ public boolean hasInitialResponse() {
             if (!MessageDigest.isEqual(decodeBase64(map.get("v")), serverSignature)) {
                 throw new SaslException("Server signature was invalid.");
             }
-            return challenge;
+            return new byte[0];
```

Comment:
ooh, nice.  I guess the server was ignoring this challenge.  Run a patch build against 4.2 to confirm

Reply:
Evergreen patch: https://evergreen.mongodb.com/version/5e348e811e2d1768a3b20ed1